### ISSUE TITLE
Storage and messaging configuration test against BE

### ISSUE THREAD

**Issue by LKCSmith** [ethyca/fides#2455]
### Is your feature request related to a specific problem?

_The work for storage and messaging configuration UI is hidden behind a feature flag until the BE components have been completed. This ticket is to test storage and messaging configuration against the live BE from the previous sprint.

### Describe the solution you'd like

_Add any related tests for the configuration to Cypress against the BE requests
_Additional user/functional testing and bug logging as discovered

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

I ( @adamsachs ) have started doing some testing on `main` now that the BE work is merged into `main`. i've taken note the following points that need to be resolved:

_Front end_

- local storage config requires at least an empty `details` nested object property -- @adamsachs has updated our confluence spec doc accordingly:
```
{
	"type": "local",
	"details": {
	},
	"format": "json"
  }
```
- `PATCH` calls to `/config` endpoint should not have an outer `fides` object in the payload, i.e. they should look just like:
```
{ 
    "storage" : {
      "active_default_storage_type" : "s3"
    }
}
```
- it seems like the details fields within the payload for storage `PUT` updates is not being populated with the values filled in by the form - here's an example request body i grabbed from my browser dev tools, even when i had auth_method and bucket populated in the UI:
```
{"type":"s3","details":{"auth_method":"","bucket":""},"format":"json"}
```
- the `notifications.notification_service_type` property in the payload for the `PATCH /config` calls are being populated with incorrect values: `mailgun-email` should be `mailgun`, `twilio-email` should be `twilio_email` and `twilio-text` should be `twilio_text`
- `PUT` calls for updating messaging config values should not include the messaging type in the URL path, i.e. it should just be `PUT /messaging/default` and the `service_type` is specified as a field in the request payload.
    - the confluence spec doc had `type` as the field in the request payload, @adamsachs has updated that to the correct `service_type` field now in the confluence doc.

_Back end_

- when a messaging `service_type` is referenced in a URL path, do not require it to be upper-cased. currently the backend requires the `GET` call to look like `http://localhost:8080/api/v1/messaging/default/MAILGUN`, which is ugly.

**Comment by LKCSmith** [comment]
This is blocked until the BE for both storage and messaging is complete.

**Comment by LKCSmith** [comment]
Please note,  #2235 and #2236 cannot be QA'd until this ticket is complete. This ticket cannot be complete until the BE work is completed. [This ticket](https://github.com/ethyca/fides/pull/2551) specifically.

**Comment by LKCSmith** [comment]
Run openAPI when this ticket is IP

**Comment by LKCSmith** [comment]
Updating points to a 3 because it looks like the API contract did have some significant changes since the UI was merged in, which means additional syncing/testing will be needed once the messaging configuration BE work is merged.

**Comment by adamsachs** [comment]
BE is now merged into `main` and i've started doing some testing. i've added the things i've noticed so far that need to be addressed to the `Additional context` section in the issue description.

**Comment by LKCSmith** [comment]
@adamsachs - I worked on some of your callouts above. You can find the draft [here](https://github.com/ethyca/fides/pull/2677). I put some TODO's in the description. If you get to your BE point above re: the URL case-sensitivity before this is merged, let me know so I can update my PR before merge.

**Comment by LKCSmith** [comment]
https://github.com/ethyca/fides/issues/2687

**Comment by adamsachs** [comment]
unfortunately on `main` i'm able to consistently reproduce the following error when on a fresh db. this occurs only the _first_ time i save the form (i.e., after saving the `API key`) for my `Mailgun Email` or `Twilio Email` messaging config. the error occurs instead of the test connection field appearing.

![image (8)](https://user-images.githubusercontent.com/9750285/223118293-c5648acc-4a1a-46a8-8c9a-d3710d0a5d7c.png)

**Comment by adamsachs** [comment]
> unfortunately on `main` i'm able to consistently reproduce the following error when on a fresh db. this occurs only the _first_ time i save the form (i.e., after saving the `API key`) for my `Mailgun Email` or `Twilio Email` messaging config. the error occurs instead of the test connection field appearing.
> 
> ![image (8)](https://user-images.githubusercontent.com/9750285/223118293-c5648acc-4a1a-46a8-8c9a-d3710d0a5d7c.png)

pushed up a potential fix for this here https://github.com/ethyca/fides/pull/2751 - @Kelsey-Ethyca  would you mind taking a look? thank you!

**Comment by Roger-Ethyca** [comment]
tested using test_env


![Image](https://user-images.githubusercontent.com/115798183/223186577-b259097b-d868-4b26-b024-6c5070fcb142.png)
![Image](https://user-images.githubusercontent.com/115798183/223186578-7a8a53e0-ada6-49be-9cf1-4331ae88198d.png)
